Spigot-Server-Patches/0619-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,201 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Fruxz <cedricspitzer@outlook.de>
+Date: Wed, 30 Dec 2020 02:23:06 +0100
+Subject: [PATCH] Added SoundBuilder
+
+
+diff --git a/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java b/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..e41b74113ea0b2b19c102dd2b1f4949d1079e4e5
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java
+@@ -0,0 +1,148 @@
++package io.papermc.paper.sound;
++
++import com.google.common.collect.Lists;
++import net.minecraft.server.BaseBlockPosition;
++import net.minecraft.server.MinecraftKey;
++import net.minecraft.server.PacketPlayOutCustomSoundEffect;
++import net.minecraft.server.Vec3D;
++import org.bukkit.Location;
++import org.bukkit.Sound;
++import org.bukkit.SoundCategory;
++import org.bukkit.World;
++import org.bukkit.craftbukkit.entity.CraftPlayer;
++import org.bukkit.entity.Player;
++import org.bukkit.util.NumberConversions;
++
++import java.util.Collection;
++
++public class PaperSoundBuilder implements SoundBuilder {
++
++    private Sound sound;
++    private SoundCategory category;
++    private Collection<Player> receivers;
```

Comment:
Shouldn't this be weak? Could leed to a memory leak  due player leaving the server and player object cannot be garbage collected.
I'm not a fan at all with persisting a collection.

Would rather say:
void play(Collection<? extends Player> players);

void play(); // is just calling eg. play(Bukkit.getOnlinePlayers()); 


Both filtering the collection of Players based on distance defined.